### ISSUE TITLE
The Future of the Transportr App (?)

### ISSUE THREAD

**Issue by Echolon** [grote/Transportr#852]
Dear Transportr Developers & Maintainers,
Dear Transportr Community,
Dear everyone interested,

I would like to raise a general topic concerning the future of this actual beautiful and very useful app that used to get me home and to my points of interest beyond my home-zone and in many countries for years.

To my understanding the development has very much declined if not stopped about [two years ago](https://github.com/grote/Transportr/commits/master) :-(
That's sad, but I also understand that there are good reasons behind it. So far I understood there was
- lack of interest from user side
- lack of support from community
- which results in a lack of motivation
- other projects I assume

Please correct me if I am wrong in any way. Of course, no blame or offense!

I would love to support changing the situation. I am not a developer per se, but I am certainly good at setting and supporting a new framework and conditions how we as a community could set a new direction of Transportr. Moreover, I see that there are also certainly [people try to up-streaming pull requests](https://github.com/grote/Transportr/pulls).

**Would you be interested? Is that something feasible? What could we do?**
I think the first steps could be to summarize the current shortcomings and get a clear picture of hurdles the development experience.

To my superficial understand I see the following points to tackle:

- Discuss if there is interest on a way forward
- If so, spread the word that we are try to start over again
- Review communication and promotion of the app
  - I believe the outstanding good idea of this software has received way to few attention in general. Communication through the Transportr webpage, [Twitter](https://twitter.com/TransportrApp) and maybe even Mastodon chould be reconsidered. The community could certainly help to upstream input here.
- Discuss and promote donations
- Consider funds alike the [NLnet](https://nlnet.nl/) or [OpenTech](https://www.opentech.fund/) fund
- Call previous contributors
- Discuss maintenance
- Discuss further development (but this is certainly not a first thing to discuss)

@grote and @ialokim many thanks for your efforts and devlepment of this app! Would you be interested in anyway to support such a discussion and reconsideration of how this could get a new chance? Do you even share my thoughts?

It would be a pleasure for me to serve and organize around this. Projects like the [StreetComplete](https://github.com/streetcomplete/streetcomplete) app managed to build a great serving community around that app.

Looking forward to revive this idea of modern public transport around the world.

Best regards,
echo

________

I know its not usually the best thing to do, but I hope it is okay that I mention previous contributors since 2022. Otherwise, I apologize.

@htgoebel 
@ashtanko
@Altonss 
@pt2121
@nomq1
@mimi89999
@1305533243
@aha999

**Acknowledgments**
Please check the following boxes with an `x` if they apply:
* [x] The feature I want to propose would be useful for the majority of users, not only for me personally.
* [x] I am aware that Transportr is mostly developed by one person in their unpaid spare time.
* [x] I can help myself to get this feature implemented or know someone who wants to do it.
* [ ] If I want to add support for a new region or country, I checked that this is already available in [public-transport-enabler](https://github.com/schildbach/public-transport-enabler/) and know the process [described on the Transportr homepage](https://transportr.app/contribute/#translating-into-your-language).

**Comment by grote** [comment]
Thanks for opening this ticket and your effort. To me the situation is quite simple: You need to find someone with the motivation, available time and skills to take over maintenance of Transportr. Once you have that, you can do more stuff, but this is really the first and most important thing you need.

**Comment by Echolon** [comment]
Thanks Torsten.

Could you specify your requirements/ expectations? Could it be multiple persons?

Furthermore, would you agree to process communication through e.g. Twitter? And maybe even pin this issues?

Cheers

16 may 2023 20:00:33 Torsten Grote ***@***.***>:

**Comment by grote** [comment]
> Could you specify your requirements/ expectations? Could it be multiple persons?

I think considering the challenge, we can't be picky here. But having the right skills to maintain an Android app will be kinda essential.

> Furthermore, would you agree to process communication through e.g. Twitter? And maybe even pin this issues?

Done.

**Comment by Echolon** [comment]
Sure, we need to see what's possible. I thought maybe listing (at least) app specific requirements may lead to that people take partly responsibility on certain parts. But maybe that's naive...

I saw you got a Mastodon account, too. Maybe post there as well? There should be a lot of open-source enthusiasts, too.

On the maintenance of the app:
So you would let a maintainer act fully self-orgainised or do you still expect to have the last word on e.g. merging? (Should a PR reviewer checklist be created?)

Thanks for your support so far!

16 may 2023 21:11:40 Torsten Grote ***@***.***>:

**Comment by ialokim** [comment]
Thanks for getting active on this one! Unfortunately, Transportr has been pretty low on my priority list in the last two years, mostly since I haven't been using the app myself. But I would definitely love to see Transportr moving forward!

The first thing to do would probably be to release a new version with the calendar fixes included. This should be pretty straightforward to prepare for someone with a working Android development environment, following the checklist (see #736) as an example.

**Comment by pt2121** [comment]
I am new to this project but I'm an experienced Android dev and I'm happy to take a stab at this and help maintain the project.

**Comment by Echolon** [comment]
Thanks ialokim for supporting this too.

When you actually plan doing another release - would there be a simple way to inform users about this thread so we can inform then. Like a basic overlay on start-up after updating. (Only if thats is not a burden)

Can I help to get this list from the issue into e.g. a MR template / readme?

16 may 2023 22:34:53 ialokim ***@***.***>:

**Comment by ialokim** [comment]
> Can I help to get this list from the issue into e.g. a MR template / readme?

It actually already is one: https://github.com/grote/Transportr/blob/master/.github/PULL_REQUEST_TEMPLATE/release.md

> basic overlay on start-up after updating

There is the changelog overlay already, some information could be included there I guess. Not sure if it supports hyperlinks though.

**Comment by Altonss** [comment]
> The first thing to do would probably be to release a new version with the calendar fixes included. This should be pretty straightforward to prepare for someone with a working Android development environment, following the checklist in the PR template (see #736 as an example).

I can try to do it :)

**Comment by Echolon** [comment]
@ialokim I don't know how much Torsten wants to be involved, but maybe you can guide a bit as "development lead" here suggesting what to go for @altonss & @prat ?

I see the situation as you said: Let's get a first small release. Btw in what file can I edit the changelog overlay? I would then push a informing message to the users there.

Next I am happy to help a bit with development prios. Maybe we actually discuss how much time people want/can to spent? We can also maybe add more places for donations for developers.
Then, a first start would be to review, test and merge open PRs, right? Especially, bugfixes and stability should be of interest I would say. Then we can call for adding more networks and features.

If thats not enough we can also put it down and apply for a fund, right?

Meaningful?

17 may 2023 12:31:35 Altonss ***@***.***>:

**Comment by Altonss** [comment]
> I see the situation as you said: Let's get a first small release.

Totally agree with this!

To me, the next steps after this would be:
- finally bring back Transportr to F-Droid, using maplibre
- then working on existing PRs/high priority bugs
- also: communicating (on Twitter, the Fediverse, other channels...) about the project

**Comment by Altonss** [comment]
> * Should we update the target version now? or, should we only release on F-Droid for now?

Publishing to F-Droid is not possible for now, see https://github.com/grote/Transportr/issues/826, so I thing the best thing to do would be to raise the target version.

**Comment by Altonss** [comment]
For the organisational part, how would you like to proceed with e.g. the labelling of issues, closing of issues, handling of PRs. Would you mind according some access to those features in the repo, maybe first only for triaging issues? This could help distribute the maintainance load!

**Comment by Echolon** [comment]
Yes that would be great. But rather let me spent time here and you can save it for more important things.

17 may 2023 21:55:50 Altonss ***@***.***>:

**Comment by Altonss** [comment]
> Yes that would be great. But rather let me spent time here and you can save it for more important things.

We can simply distribute the tasks among different people :)

**Comment by Echolon** [comment]
Of course, I just thought I take over the non-technical stuff, as I barely can support on development directly :-/

17 may 2023 22:07:30 Altonss ***@***.***>:

**Comment by pt2121** [comment]
@grote @ialokim I opened a PR to update the target SDK version since Google Play now requires 31+. Can you take a look when having a chance?
Also, I have some other questions [here](https://github.com/grote/Transportr/issues/852#issuecomment-1551918703).
@Echolon @Altonss maybe we can discuss about the process and work distribution in [discussions](https://github.com/grote/Transportr/discussions) or in another thread as needed. (I just think this one is getting long)

**Comment by Echolon** [comment]
Hey Prat ***@***.*** ! :-)),

I think its a good idea with discussions. just lets not move tracking and discussing of issues there. Rather general organisation, right?

From my side please go ahead with creating threads on the first next release publication if thats what you were looking for.

Cheers

18 may 2023 6:16:28 Prat ***@***.***>:

**Comment by Altonss** [comment]
For 2.1.5 release, the thread is here: https://github.com/grote/Transportr/discussions/854

**Comment by ialokim** [comment]
> The latest PTE is [broken on jitpack](https://jitpack.io/#com.gitlab.opentransitmap/public-transport-enabler).

The PTE repo we use also happens to be maintained by me, I will try to find some time to look into it.

**Comment by Altonss** [comment]
> The PTE repo we use also happens to be maintained by me, I will try to find some time to look into it.

No worries, take your time :) (It's a project in our free time for all of us here)

**Comment by Echolon** [comment]
I started a general discussion on a [vision for the app here](https://github.com/grote/Transportr/discussions/858)

**Comment by Pegzik** [comment]
Hey @Echolon, just wanted to say that I'm interested in supporting your idea. I'm not really a developer nor have any real coding experience, but I'd like to help & learn in whichever way if possible. Is there any way of easy contact like via e-mail, perhaps Discord or something similar?

Thank you as well as the whole team that has been working on this project for years for trying to keep this app alive!

**Comment by Echolon** [comment]
Hey @Pegzik,

great you are interested. I'd suggest to follow the threads for the moment and maybe share your ideas. Right now the developers try to get a fixing release. Then let's what comes next. I think you can help best to spread the word currently.

In terms of communication I only use XMPP. I suggest we use the Github Discussion threads for the moment to keep communication very open.

Cheers

21 may 2023 15:34:27 Pegzik ***@***.***>:

**Comment by SandResidence** [comment]
Hello everyone,

I apologize in advance if any part of this post is incomprehensible.
English is not my native language.

Some time ago, I started moving away from closed-source applications (such as Google Maps) to open-source alternatives developed by the community (such as Transportr).

From my several days of searching, Transportr is the only app that declares to support my region (Poland) and is likely to be updated (other apps haven't received any commits for several years)

I have no experience in programming android apps (I only know the basics of Java), but I would like to help develop this app as much as I can.

I would like to further ask if you are perhaps planning to change the design of the app in the near future? I want all the apps I have to have Material Design 3 (which looks AMAZING) and support for Material You color synchronization.

**Comment by Echolon** [comment]
> Hello everyone,
> 
> I apologize in advance if any part of this post is incomprehensible. English is not my native language.
> 
> Some time ago, I started moving away from closed-source applications (such as Google Maps) to open-source alternatives developed by the community (such as Transportr).
> 
> From my several days of searching, Transportr is the only app that declares to support my region (Poland) and is likely to be updated (other apps haven't received any commits for several years)
> 
> I have no experience in programming android apps (I only know the basics of Java), but I would like to help develop this app as much as I can.
> 
> I would like to further ask if you are perhaps planning to change the design of the app in the near future? I want all the apps I have to have Material Design 3 (which looks AMAZING) and support for Material You color synchronization.

Hey,

great to hear! I recommend you check if you ideas or suggestions exist already, otherwise you can describe it in new issues.

Other then that you you can of course spread the word or donate to the developers and volunteers of course.

Maybe join also the vision discussion.

**Comment by atollk** [comment]
Hey. There seems to be no official conclusion in this issue thread and the commit history has been slow in recent times. I see that some PRs have been reviewed but does that mean there is still no active maintainer?

**Comment by atollk** [comment]
I have some experience in developing web apps, including Kotlin, and wanted to give Android app development a try, to a yet uncertain degree. But I have had bad experiences with contributing to open source projects that have been abandoned by their maintainer, as they always died at some point and my work went to waste. Especially since even beyond the Github repo ownership, there seem to be other points of contact such as the donations and JawgMaps.

**Comment by GenEmSec** [comment]
Having found this discussion after not being able to query Deutsche Bahn routes for some time now (API changed by DB), I do wonder - is anyone actively maintaining the app?

It's not to put pressure on, and I'm grateful for anyone's work and I also understand that most of the sensible stuff is done after work at night (same place I'm in). So I get it if one's overwhelmed and slower than one would like - it's only I'd like to know whether the switch back to Öffi (which I don't like as much - no offense 😅) is looking to be permanent, or whether there's an update round the corner...

**Comment by bernhardreiter** [comment]
> not being able to query Deutsche Bahn routes for some time now (API changed by DB)

See #1019 for more details about this. It seems all is missing is a new transportr release.

> or whether there's an update round the corner...

There has not been a release since November 2025, so more help is needed.
